### PR TITLE
[IMP] *: views: deprecate active_* keys from evalContext

### PR DESCRIPTION
**Pull request by aab-odoo**
This commit aims to simplify the evaluation context used to
evaluate expressions used in views (invisible, required, readonly,
domain and context attributes). For now, the evaluation context is
typically the current record (there's a key for each field in the
view). In addition to that, there're static keys (that may conflict
with field names): uid, allowed_company_ids, current_company_id,
active_id, active_ids and active_model.

The motivation of this commit is at some point to get rid of the
3 active_* keys, because they are misleading and basically useless.

The notion of active_* exists, but it is something else: when you
are in a form view (let's say the form of a partner) and you open
its opportunities (by clicking on the stat button), the list view
of opportunities shows up and in the context, there're 3 keys
active_* , referring to the record from which we came. One can
easily access those information with context.get("active_*"), in
python or in view archs.

However, almost all `active_id` found in archs were actually used
to refer to the id of the current record. Indeed, for now, in the
evaluation context of a record, the value of the `active_id` key is
always the id of the record. So this commit adapts them to
directly use `id` instead. There was no use of active_ids, and
a single use of active_model which was removed (active_model is
the res_model of the view, so it isn't really necessary).

This commit doesn't drop the support of those keys, it deprecates
them. They will be removed for v18. A warning will be displayed if
they are used.

### CODE REVIEW

File: addons/purchase/views/product_views.xml [hunk 1]
```diff
@@ -77,6 +77,9 @@
                 <field name="seller_ids" position="attributes">
                     <attribute name="context">{'default_product_tmpl_id': product_tmpl_id, 'product_template_invisible_variant': True, 'tree_view_ref':'purchase.product_supplierinfo_tree_view2'}</attribute>
                 </field>
+                <field name="variant_seller_ids" position="attributes">
+                    <attribute name="context">{'model': 'product.product', 'active_id': id, 'tree_view_ref':'purchase.product_supplierinfo_tree_view2'}</attribute>
```
Comment (aab-odoo) [inline addons/purchase/views/product_views.xml:81]: is it really necessary?

File: addons/lunch/views/lunch_orders_views.xml [hunk 2]
```diff
@@ -274,8 +274,7 @@
                     </div>
                 </div>
                 <footer>
-                    <button string="Save" special="save" data-hotkey="q" class="oe_highlight" invisible="not context.get('active_id', False)"/>
-                    <button string="Add To Cart" name="add_to_cart" type="object" class="oe_highlight" invisible="context.get('active_id', False) or order_deadline_passed" data-hotkey="w"/>
+                    <button string="Add To Cart" name="add_to_cart" type="object" class="oe_highlight" invisible="order_deadline_passed" data-hotkey="w"/>
```
Comment (aab-odoo) [inline addons/lunch/views/lunch_orders_views.xml:277]: double check

File: addons/web/static/src/model/relational_model/static_list.js [hunk 3]
```diff
@@ -90,6 +90,7 @@ export class StaticList extends DataPoint {
             context,
             uid: context.uid,
             allowed_company_ids: context.allowed_company_ids,
+            // add current_company_id
```
Comment (aab-odoo) [inline addons/web/static/src/model/relational_model/static_list.js:93]: todo

File: addons/account/views/account_journal_dashboard_view.xml [hunk 4]
```diff
@@ -107,7 +107,7 @@
                                         <span role="separator">Reconciliation</span>
                                     </div>
                                     <div>
-                                        <a role="menuitem" type="object" name="open_action_with_context" context="{'action_name': 'action_account_reconcile_model', 'use_domain': ['|', ('match_journal_ids', '=', False), ('match_journal_ids', 'in', active_id)]}" groups="account.group_account_manager">Reconciliation Models</a>
+                                        <a role="menuitem" type="object" name="open_action_with_context" context="{'action_name': 'action_account_reconcile_model', 'use_domain': ['|', ('match_journal_ids', '=', False), ('match_journal_ids', 'in', id)]}" groups="account.group_account_manager">Reconciliation Models</a>
```
Comment (Gorash) [inline addons/account/views/account_journal_dashboard_view.xml:110]: `('match_journal_ids', 'in', id)`
did it work before?
Shouldn't it be `[id]`?

**Review by rco-odoo — APPROVED** [review]
Looks fine to me.

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**